coursera/api.py

```diff
@@ -755,9 +763,16 @@ def extract_links_from_programming(self, element_id):
 
         # Assignment text (instructions) contains asset tags which describe
         # supplementary files.
-        text = ''.join(self._extract_assignment_text(element_id))
-        if not text:
-            return {}
+        try:
+            text = ''.join(self._extract_assignment_text(element_id))
+            if not text:
+                return {}
+        except HTTPError as ex:
+            FORBIDDEN = 403
```

Comment:
Why are you introducing the same constant over and over again? Define it once and use it everywhere. But that's a side note because you shouldn't filter only 403 as I mentioned above.